### ISSUE TITLE
enforce-canonical: false positive — rounded-[4px] → rounded-lg when --radius is overridden in :root

### ISSUE THREAD

**Issue by mindtraveller** [sergioazoc/oxlint-tailwindcss#78]
## Description

The `tailwindcss/enforce-canonical` rule incorrectly suggests converting `rounded-[4px]` to `rounded-lg` (and `rounded-[4px]` to `rounded-sm`) on projects that override the `--radius` CSS variable in `:root`.

This is the standard shadcn/ui pattern for customizing border radius — the project's `src/index.css` contains:

```css
@theme inline {
  --radius-sm: calc(var(--radius) - 4px);
  --radius-md: calc(var(--radius) - 2px);
  --radius-lg: var(--radius);
  --radius-xl: calc(var(--radius) + 4px);
}

:root {
  --radius: 0.875rem;  /* 14px — overrides Tailwind's default 0.25rem (4px) */
}
```

With this override, `rounded-lg` resolves to 14px at runtime, **not** 4px. But `enforce-canonical` still maps `rounded-[4px]` to `rounded-lg`, which silently changes the visual design (3.5x larger corners).

## Root cause

The rule delegates to Tailwind v4's `canonicalizeCandidates()` API. I verified the behavior by calling `ds.candidatesToCss()` directly:

```js
const { __unstable__loadDesignSystem } = require('@tailwindcss/node');
const css = readFileSync('src/index.css', 'utf-8');
const ds = await __unstable__loadDesignSystem(css, { base: 'src' });

ds.candidatesToCss(['rounded-lg']);
// → .rounded-lg { border-radius: var(--radius); }

ds.candidatesToCss(['rounded-[4px]']);
// → .rounded-\[4px\] { border-radius: 4px; }
```

The CSS outputs are **not** equivalent (`var(--radius)` = 14px vs `4px` literal), yet `canonicalizeCandidates(['rounded-[4px]'])` returns `rounded-lg` because it uses Tailwind's **default** `--radius` value (0.25rem = 4px) for the match, not the project's `:root` override.

## Impact

8 incorrect autofixes were silently applied across 6 files in our codebase before we caught it. This affects every shadcn/ui project that customizes `--radius` — which is the documented shadcn pattern.

## Environment

- `oxlint-tailwindcss`: v1.3.5
- `tailwindcss`: v4.3.2
- `@tailwindcss/node`: v4.3.2

## Expected behavior

`enforce-canonical` should not suggest `rounded-[4px]` to `rounded-lg` when the project's `--radius-lg` token does not resolve to 4px. If `canonicalizeCandidates` cannot resolve `:root` CSS variable overrides, the rule should either:

1. Skip canonicalization for `rounded-[Npx]` arbitrary values entirely, or
2. Compare the actual CSS output (like the `arbitraryEquivalents` precompute already does for other properties) instead of relying solely on `canonicalizeCandidates`.

## Workaround

We disabled the rule:

```json
"tailwindcss/enforce-canonical": "off"
```

## Minimal reproduction

Any project with:

```css
@theme inline {
  --radius-lg: var(--radius);
}
:root {
  --radius: 0.875rem;
}
```

Then `enforce-canonical` will flag `rounded-[4px]` as non-canonical and suggest `rounded-lg`, which resolves to 14px instead of 4px.

**Comment by sergioazoc** [comment]
Fixed by #82 (merged): enforce-canonical now only rewrites an arbitrary value to a named token when the two forms emit byte-identical CSS. A literal like `rounded-[4px]` is no longer autofixed to a variable-backed token (`rounded-lg` = `var(--radius-lg)`) whose value a `:root` override changes — the silent design corruption is gone. Value-preserving conversions (legacy renames, var-syntax normalization, literal-valued utilities) still apply. Thanks for the precise root-cause analysis; the `candidatesToCss` comparison you suggested is exactly what the fix keys on.
